core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2331,7 +2331,7 @@ private void verify(RexNode before, RexNode simplified, RexUnknownAs unknownAs)
 
   private RexNode simplifySearch(RexCall call, RexUnknownAs unknownAs) {
     assert call.getKind() == SqlKind.SEARCH;
-    final RexNode a = call.getOperands().get(0);
+    final RexNode a = simplify(call.getOperands().get(0), unknownAs);
```

Comment:
In general I find it more convenient to have one function call per line - makes it much easier to use the debugger to see what's going on. So I think `simplify(a)` should be a separate statement.